plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -844,4 +881,33 @@ public interface TransportSupplier
         TTransport createTransport()
                 throws TTransportException;
     }
+
+    /**
+     * To construct a pattern using database and catalog name that Hive Thrift Server.
+     * Based on the Hive's <a href="https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java">implementation</a>.
```

Comment:
i'd advocate here for a tag or specific commit in referencing `MetaStoreUtils.java` - in `master` the class may be changed/moved in the future.